Update German translations (#241)

rysk-de opened the pull request.

DamascenoRafael commented on reminders-menubar/Resources/Localizable.xcstrings:
```diff
@@ -1331,7 +1337,7 @@
         "de" : {
           "stringUnit" : {
-            "state" : "needs_review",
+            "state" : "translated",
             "value" : "Verschieben nach ..."
```
Initial text for this key in English was "Move to...", but now the text is "List" which is more aligned with what is presented in Apple Reminders (image below).

Can you re-check what would be the most appropriate translation considering the English text is "List"?

<img width="686" alt="image" src="https://github.com/user-attachments/assets/72c33bfd-a00d-49b1-8ae5-ba8b11e8d8bf" />

DamascenoRafael reviewed (changes requested):
Thanks @rysk-de. Could you please check the minor comment I added?

rysk-de commented:
> Thanks @rysk-de. Could you please check the minor comment I added?

Hi @DamascenoRafael, good catch! I changed the mentioned translation and found another one at line 1620 that did not represent the wording in Apple Reminders.

DamascenoRafael commented:
Awesome, thanks for helping with the translations @rysk-de!
